helix-term/src/ui/completion.rs

```diff
@@ -323,20 +326,41 @@ impl Completion {
                             doc.apply(&transaction, view.id);
                         }
                     }
+                    // we could have just inserted a trigger char (like a `crate::` completion for rust
+                    // so we want to retrigger immedietly when accepting a completion.
+                    trigger_auto_completion(&editor.handlers.completions, editor, true);
                 }
             };
+
+            // In case the popup was deleted because of an intersection w/ the auto-complete menu.
+            if event != PromptEvent::Update {
+                editor
+                    .handlers
+                    .trigger_signature_help(SignatureHelpInvoked::Automatic, editor);
+            }
         });
         let popup = Popup::new(Self::ID, menu)
             .with_scrollbar(false)
             .ignore_escape_key(true);
+        let (view, doc) = current_ref!(editor);
+        let cursor = doc
+            .selection(view.id)
+            .primary()
+            .cursor(doc.text().slice(..));
+        let fragment = doc.text().slice(start_offset..cursor);
```

Comment:
I just had a panic here. But I'm not sure how it did happen exactly (reproduction):
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', helix-term/src/ui/completion.rs:350:24
```
Otherwise solid PR! (Skimmed a little bit through the code.)

Reply:
That was a really good catch, thanks! I think the reason here was that I calculated `start_offset` before requesting the completion (which could be incorrect since it was used to read the current document). This could lead to incorrect filtering with multi cursors and in some (very rare) cases to crashes (if text was deleted/the primary cursor was moved in insert mode right after the request finished). 

I moved the `start_offset` calculation right into the UI component so it is now trivially <= cursor (and should actually complete the fix for a logic bug I found in master it just didn't lead to panics in the past)